codersdk/client.go

```diff
@@ -171,8 +171,8 @@ func readBodyAsError(res *http.Response) error {
 
 // Error represents an unaccepted or invalid request to the API.
 // @typescript-ignore Error
-type Error struct {
-	httpapi.Response
+type HTTPError struct {
```

Comment:
I think this should still be `Error`. We don't prefix `Response` with HTTP, so this shouldn't either IMO.

Reply:
https://github.com/coder/coder/pull/2954/files#diff-7fad09d9d87a67ca91331d8399caa8b6644a66ae3b5fec5534cdeeff06a4b881R24

there's a conflict in naming when I moved this in, which is why i updated the name. I think it's appropriate given it contains http-specific fields

Reply:
I see the diff now. I'd rather rename the other to `ValidationError` in that case. Seems more apt